src/value.rs

```diff
@@ -113,6 +120,99 @@ impl<'gc> Value<'gc> {
         }
     }
 
+    // A small helper function to handle division-like zero handling
+    fn safe_div<T: PartialEq + ToPrimitive + Zero>(
+        lhs: T,
+        rhs: T,
+        f: &Fn(T, T) -> Value<'gc>,
+    ) -> Value<'gc> {
+        match (lhs, rhs) {
+            // Seems that all nans are negative in lua
+            (ref a, ref b) if a.is_zero() && b.is_zero() => Value::Number(-f64::NAN),
+            (ref a, ref b) if b.is_zero() => {
+                Value::Number(copysign(f64::INFINITY, a.to_f64().unwrap()))
+            }
+            (a, b) => f(a, b),
+        }
```

Comment:
This is prooobably not best expressed as a match, `if a.is_zero() && b.is_zero() { ... } else if b.is_zero() { ... } else { ... }` is better.

Reply:
That's fair.